cmd/otel-allocator/prehooktargetfilter/relabel_config_filter.go

```diff
@@ -0,0 +1,101 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package prehooktargetfilter
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/go-logr/logr"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/model/relabel"
+
+	"github.com/open-telemetry/opentelemetry-operator/cmd/otel-allocator/allocation"
+)
+
+type RelabelConfigTargetFilter struct {
+	m           sync.RWMutex
+	log         logr.Logger
+	targetItems map[string]*allocation.TargetItem
+	allocator   allocation.Allocator
+}
+
+func NewRelabelConfigTargetFilter(log logr.Logger, allocator allocation.Allocator) AllocatorPrehook {
+	return &RelabelConfigTargetFilter{
+		log:         log,
+		allocator:   allocator,
+		targetItems: make(map[string]*allocation.TargetItem),
+	}
+}
+
+func (tf *RelabelConfigTargetFilter) SetTargets(targets map[string]*allocation.TargetItem) {
+	numRemainingTargets := 0
+	for jobName, tItem := range targets {
+		keepTarget := true
+		for _, cfg := range tItem.RelabelConfigs {
+			if tf.IsDropTarget(tItem.Label, cfg) {
+				keepTarget = false
+				break
+			}
+		}
+
+		if keepTarget {
+			tf.targetItems[jobName] = tItem
+			numRemainingTargets++
+		}
+	}
+
+	tf.allocator.SetTargets(tf.targetItems)
+	tf.log.Info(fmt.Sprintf("Relabel filtering completed. Keeping %d target(s) out of %d total targets", numRemainingTargets, len(targets)))
```

Comment:
This might be a bit much to emit at the `info` level.  Doesn't this get run every time the discovery manager runs any of its sources?

Also, don't use `fmt.Sprintf` for logging, we're using a structured logger and should take advantage of it:

```go
tf.log.V(30).Info("Filtering complete", "seen", len(targets), "kept", numRemainingTargets)
```

Not sure what the appropriate `V()` value is here.  Are there standard values used elsewhere in the operator codebase?

Reply:
Yes you're correct, this was just helpful for my debugging purposes while testing. A simple grep of the repo shows V(1) and V(2) used. So maybe for now I should use V(2) for the verbosity? Let me know if you have other preferences.